client/src/components/StoreList.tsx

```diff
@@ -117,18 +123,73 @@ function StoreList() {
     }
   };
 
+  // Scan Store's QR code
+  const storeQR = async () => {
+    if (window.location == window.parent.location) {
+      const img = document.getElementById(
+        "sampleStoreQRImgSrc"
+      ) as HTMLImageElement;
+      processImageBarcode(img);
+    } else {
+      const imgReq = {
+        allowedMimeTypes: ["image/jpeg"],
+        allowedSources: ["camera"], // Restrict to camera scanning only
+      };
+      const imgRes = await window.microapps
+        .requestMedia(imgReq)
+        .then((res: any) => res);
+      setUploadImg(imgRes);
+    }
+  };
+
+  const processImageBarcode = async (img: HTMLImageElement) => {
+    const codeReader = new BrowserMultiFormatReader();
+    if (img != null) {
+      const result = await codeReader
+        .decodeFromImage(img)
+        .then((res: any) => res)
+        .catch((err: any) => {
+          console.log(err);
+          return "";
+        });
+      if (result) {
+        window.location.href = "/store?" + result.text;
+      } else {
+        console.log("Unable to redirect, QR Code not scanned correctly");
+      }
+    } else {
+      console.log("Cannot find element: storeQRImgSrc");
+    }
+  };
+
+  useEffect(() => {
+    if (uploadImg.mimeType) {
+      const img = document.getElementById("storeQRImgSrc") as HTMLImageElement;
+      processImageBarcode(img);
+    }
+  }, [uploadImg]);
+
   return (
     <div className="StoreList">
+      {uploadImg.mimeType && (
```

Comment:
I think the image upload logic is certainly complex and should be extracted out to a smaller react component and called inside the StoreList.

Reply:
TODO #66